packages/sql-mapper/lib/entity.js

```diff
@@ -120,8 +120,10 @@ function createMapper (db, sql, log, table, fields, primaryKey, relations, queri
       return Object.values(inputToFieldMap)
     }
 
+    const relationalFields = fields.filter((field) => alwaysRetrieve.includes(field))
     const requestedFields = fields.map((field) => inputToFieldMap[field])
```

Comment:
Or perhaps:

```js
const requestedFields = fields.map((field) => inputFieldMap[field] || field)
```

Reply:
Cannot just pass the field, but I did refactor to single map instead of filter and map. Needed to check if in the relations of current entity.